examples/plot_digits.py

```diff
@@ -0,0 +1,208 @@
+ward_clustering_1_7.fit(X_cluster)
+
+ward_clustering_0_1 = FeatureAgglomeration(
+    n_clusters=n_clusters, connectivity=connectivity
+)
+X_cluster = train_test_split(X_0_1, train_size=0.5, random_state=0)[0]
+ward_clustering_0_1.fit(X_cluster)
+
+_, axes = plt.subplots(1, 2, figsize=(6, 3))
+ax = axes[0]
+ax.imshow(ward_clustering_1_7.labels_.reshape(28, 28), cmap="Set2")
+ax.axis("off")
+_ = ax.set_title("Clustering for digits 1 vs 7")
+
+ax = axes[1]
+ax.imshow(ward_clustering_0_1.labels_.reshape(28, 28), cmap="Set2")
+ax.axis("off")
+_ = ax.set_title("Clustering for digits 0 vs 1")
+
+_ = plt.tight_layout()
+
+
+# %%
+# Randomness in clustering affects inference stability
+# ----------------------------------------------------
+# A key limitation of clustered inference procedures is that the clustering step
+# introduces randomness into the inference process. Data perturbations can produce
+# different clustering results, which subsequently lead to varying inference outcomes.
+# This effect can be demonstrated by running the clustering algorithm multiple times
+# on different subsamples of the data.
+
+
+_, axes = plt.subplots(1, 4, figsize=(9, 3))
+
+for i, ax in enumerate(axes):
+    X_cluster = train_test_split(X_1_7, y_1_7, train_size=0.5, random_state=i)[0]
+    ward_clustering = FeatureAgglomeration(
+        n_clusters=n_clusters, connectivity=connectivity
+    )
+    ward_clustering.fit(X_cluster)
+    ax.imshow(ward_clustering.labels_.reshape(28, 28), cmap="Set2")
+    ax.axis("off")
+    ax.set_title(f"Clustering {i}")
+
+_ = plt.tight_layout()
+
+
+# %%
+# Ensemble Clustered Inference with Desparsified Lasso
+# ----------------------------------------------------
+# To mitigate the randomness introduced by clustering, we ensemble the results from
+# multiple clustered inference procedures. This approach derandomizes the inference
+# procedure and produces more stable results. We use the `ensemble_clustered_inference`
+# function for this purpose. While this approach is more computationally intensive than
+# single clustered inference, the procedure can be parallelized across clustering
+# repetitions using the `n_jobs` parameter.
+
+from sklearn.preprocessing import StandardScaler
+
+from hidimstat import ensemble_clustered_inference
+from hidimstat.ensemble_clustered_inference import ensemble_clustered_inference_pvalue
+
+n_jobs = 5
+fdr = 0.5
+
+ward_1_7, desparsified_lassos_1_7 = ensemble_clustered_inference(
+    X_1_7,
+    y_1_7,
+    ward_clustering,
+    scaler_sampling=StandardScaler(),
+    random_state=0,
+    n_jobs=n_jobs,
+)
+ward_0_1, desparsified_lassos_0_1 = ensemble_clustered_inference(
+    X_0_1,
+    y_0_1,
+    ward_clustering,
+    scaler_sampling=StandardScaler(),
+    random_state=0,
+    n_jobs=n_jobs,
+)
+
+beta_hat_1_7, selected_ecdl_1_7 = ensemble_clustered_inference_pvalue(
+    X_1_7.shape[0], False, ward_1_7, desparsified_lassos_1_7, fdr=fdr
```

Comment:
I'd rather use fwer control rather than fdr control. I think it makes more sense for EncluDL